app/src/features/apiClient/contexts/apiClient.tsx

```diff
@@ -178,21 +176,22 @@ export const ApiClientProvider: React.FC<ApiClientProviderProps> = ({ children }
   const onSaveRecord = useCallback(
     (apiClientRecord: RQAPI.Record) => {
       const isRecordExist = apiClientRecords.find((record) => record.id === apiClientRecord.id);
-
+      const urlPath = apiClientRecord.type === RQAPI.RecordType.API ? "request" : "collection";
       if (isRecordExist) {
         onUpdateRecord(apiClientRecord);
+        updateTab(apiClientRecord.id, {
+          title: apiClientRecord.name,
+          url: `${PATHS.API_CLIENT.ABSOLUTE}/${urlPath}/${apiClientRecord.id}`,
+        });
```

Comment:
`onUpdateRecord` handles the already handles updateTab [refer](https://github.com/requestly/requestly/blob/bdee197b65e9364f06b46c2cdafddd0077020dee/app/src/features/apiClient/contexts/apiClient.tsx#L154)